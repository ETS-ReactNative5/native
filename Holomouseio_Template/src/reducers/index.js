import { changeAppState } from "./appStates";
import { changeSelectionType } from "./selectionTypes";

export default reducers = (state = {}, action) => {
    return {
        appState: changeAppState(state.appState, action),
        selectionType: changeSelectionType(state.selectionType, action)
    }
}