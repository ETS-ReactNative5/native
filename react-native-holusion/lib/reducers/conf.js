'use strict';
import { SET_CONF, SET_SLIDES_CONTROL, SET_DEFAULT_TARGET, SET_PURGE, SET_PROJECTNAME} from '../actions'

export default function conf(state = {
    projectName: undefined, 
    configurableProjectName: false, // This is read-only from initialization
    default_target: null,
    purge_products: false,
    slides_control: "default",
}, action) {
    switch(action.type) {
        case SET_CONF:
            return Object.assign({}, state, action.conf);
        case SET_DEFAULT_TARGET:
            return Object.assign({}, state, {default_target: action.target});
        case SET_PURGE:
            return Object.assign({}, state, {purge_products: action.purge});
        case SET_SLIDES_CONTROL:
            return Object.assign({}, state, {slides_control: action.controlType});
        case SET_PROJECTNAME:
            if(!state.configurableProjectName){
                console.warn("Trying to modify read-only project name");
                return state;
            }
            return Object.assign({}, state, {projectName: action.name});
        default:
            return state;
    }
}