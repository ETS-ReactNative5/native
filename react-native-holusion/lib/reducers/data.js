'use strict';
import { SET_DATA, SET_CONFIG, SET_SLIDES_CONTROL} from '../actions'

export default function data(state = {
    items:{},
    config:{}, 
    projectName:null, 
    userName: null, 
    password: null,
    slides_control: "default",
}, action) {
    switch(action.type) {
        case SET_SLIDES_CONTROL:
            return Object.assign({}, state, {slides_control: action.controlType});
        case SET_DATA:
            return Object.assign({}, state, action.data);
        case SET_CONFIG:
            return Object.assign({}, state, {config: action.config});
        default:
            return state;
    }
}