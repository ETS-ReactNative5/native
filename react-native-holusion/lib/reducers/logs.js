'use strict';

export default function data(state = {
    logs:[],
}, action) {
  if(action.error){
    return [...state.logs, {severity: "error", message: action.error.message, timestamp: new Date()}]
  }
  return state;
}