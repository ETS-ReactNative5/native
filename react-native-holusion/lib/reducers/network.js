'use strict';
import { SET_NETINFO, SET_FIREBASEINFO } from '../actions'

export default function network  (state = {status: "offline", firebase: "disconnected"}, action) {
    switch(action.type) {
        case SET_NETINFO:
            return Object.assign({},state, {status: action.status});
        default:
            return state;
    }
}