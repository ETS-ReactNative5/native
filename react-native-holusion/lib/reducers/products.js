'use strict';
import { SET_PRODUCTS, SET_ACTIVE, ADD_PRODUCT, REMOVE_PRODUCT} from '../actions'

export default function products  (state = [], action) {
    switch(action.type) {
        case SET_PRODUCTS:
            return action.products;
        case SET_ACTIVE:
            const products = state.map(p =>{
                if(p.name == action.name){
                    return Object.assign(p, {active: true});
                }else{
                    return Object.assign(p, {active: false})
                }
            })
            return products;
        case ADD_PRODUCT:
            return [action.product].concat(state);
        case REMOVE_PRODUCT:
            return state.filter(elem => elem.name != action.name)
        default:
            return state;
    }
}