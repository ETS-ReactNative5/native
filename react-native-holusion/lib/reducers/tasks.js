'use strict';
import { SET_NETINFO, SET_FIREBASEINFO, ADD_TASK, REMOVE_TASK, UPDATE_TASK } from '../actions'

export default function tasks(state = { list: {} }, {type, id, ...action}) {
  let list;
  switch (type) {
    case UPDATE_TASK:
      return Object.assign({}, state, {
        list: {
          ...state.list,
          [id]: Object.assign({}, state.list[id], action),
        },
      });
    case ADD_TASK:
      return Object.assign({}, state, {
        list: {
          ...state.list,
          [id]: action,
        },
      });

    case REMOVE_TASK:
      const {[id]: _, ...list} = {...state.list};
      return Object.assign({}, state, {
        list,
      });
    default:
      return state;
  }
}