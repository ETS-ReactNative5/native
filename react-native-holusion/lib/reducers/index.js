'use strict';
import products from "./products";
import network from "./network";
import data from "./data";
import tasks from "./tasks";
import conf from "./conf";
import logs from "./logs";

import {combineReducers} from "redux";
export default combineReducers({
  products,
  network,
  data,
  tasks,
  conf,
  logs,
});