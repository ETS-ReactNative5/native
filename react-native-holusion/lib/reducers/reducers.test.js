import reducers from ".";

describe("reducers", function(){
  test("initializes reducers with default state", ()=>{
    expect(typeof reducers).toBe("function");
    expect(reducers(undefined, {})).toMatchSnapshot();
  })
})