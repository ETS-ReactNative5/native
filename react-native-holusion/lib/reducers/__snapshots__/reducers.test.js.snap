// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers initializes reducers with default state 1`] = `
Object {
  "conf": Object {
    "configurableProjectName": false,
    "default_target": null,
    "play_control": "none",
    "projectName": undefined,
    "purge_products": false,
    "slides_control": "default",
  },
  "data": Object {
    "config": Object {},
    "items": Object {},
  },
  "network": Object {
    "firebase": "disconnected",
    "status": "offline",
  },
  "products": Array [],
  "tasks": Object {
    "list": Object {},
  },
}
`;
